src/rust/rust-proto-new/tests/test_utils/strategies.rs

```diff
@@ -120,6 +120,396 @@ pub mod pipeline {
+            ImmutableUintProp {
+                prop
+            }
+        }
+    }
+
+    //
+    // IncrementOnlyIntProp
+    //
+
+    prop_compose! {
+        pub fn increment_only_int_props()(
+            prop in any::<i64>(),
+        ) -> IncrementOnlyIntProp {
+            IncrementOnlyIntProp {
+                prop
+            }
+        }
+    }
+
+    //
+    // IncrementOnlyUintProp
+    //
+
+    prop_compose! {
+        pub fn increment_only_uint_props()(
+            prop in any::<u64>(),
+        ) -> IncrementOnlyUintProp {
+            IncrementOnlyUintProp {
+                prop
+            }
+        }
+    }
+
+    //
+    // Edge
+    //
+
+    prop_compose! {
+        pub fn edges()(
+            to_node_key in any::<String>(),
+            from_node_key in any::<String>(),
+            edge_name in any::<String>(),
+        ) -> Edge {
+            Edge {
+                to_node_key,
+                from_node_key,
+                edge_name,
+            }
+        }
+    }
+
+    //
+    // EdgeList
+    //
+
+    prop_compose! {
+        pub fn edge_lists()(
+            edges in collection::vec(edges(), 10),
+        ) -> EdgeList {
+            EdgeList {
+                edges
+            }
+        }
+    }
+
+    //
+    // Session
+    //
+
+    prop_compose! {
+        pub fn sessions()(
+            primary_key_properties in collection::vec(any::<String>(), 10),
+            primary_key_requires_asset_id in any::<bool>(),
+            create_time in any::<u64>(),
+            last_seen_time in any::<u64>(),
+            terminate_time in any::<u64>(),
+        ) -> Session {
+            Session {
+                primary_key_properties,
+                primary_key_requires_asset_id,
+                create_time,
+                last_seen_time,
+                terminate_time
+            }
+        }
+    }
+
+    //
+    // Static
+    //
+
+    prop_compose! {
+        pub fn statics()(
+            primary_key_properties in collection::vec(any::<String>(), 10),
+            primary_key_requires_asset_id in any::<bool>(),
+        ) -> Static {
+            Static {
+                primary_key_properties,
+                primary_key_requires_asset_id
+            }
+        }
+    }
+
+    //
+    // Strategy
+    //
+
+    pub fn strategies() -> impl Strategy<Value = GraphStrategy> {
+        prop_oneof![
+            sessions().prop_map(|session| GraphStrategy::Session(session)),
+            statics().prop_map(|static_| GraphStrategy::Static(static_)),
+        ]
+    }
+
+    //
+    // IdStrategy
+    //
+
+    prop_compose! {
+        pub fn id_strategies()(
+            strategy in strategies()
+        ) -> IdStrategy {
+            IdStrategy { strategy }
+        }
+    }
+
+    //
+    // Property
+    //
+
+    pub fn properties() -> impl Strategy<Value = Property> {
+        prop_oneof![
+            decrement_only_int_props().prop_map(|p| Property::DecrementOnlyIntProp(p)),
```

Comment:
FWIW you can do `.prop_map(Property::DecrementOnlyIntProp)`
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=fd8d5f3fe7a1a595cffc8563e1a6508c

Reply:
done